module/zfs/dmu_objset.c

```diff
@@ -1748,30 +1811,17 @@ dmu_objset_sync(objset_t *os, zio_t *pio, dmu_tx_t *tx)
 		sda->sda_list = ml;
 		sda->sda_sublist_idx = i;
 		sda->sda_tx = tx;
+		sda->sda_cio = zio_null(sio, os->os_spa, NULL, NULL, NULL,
```

Comment:
This is part of replacing the taskq_wait() so that the current thread does not need to block. The sio is the parent and we will zio_nowait on that, a few lines below.  The zio_nowait(sda->sda_cio) is done from the sync_dnodes_task() which is dispatched to the dp_sync_taskq.

Reply:
It might be worth a comment, as it does look a little weird to use zios as wait points for threads, rather than actual IO. But I think its a neat solution to an awkward problem, since its not easy to hook up the real work zios, which are stashed in the dnodes and already hooked up.

Reply:
code updated per conference discussions.